crates/ruff_linter/src/rules/ruff/rules/sequence_sorting.rs

```diff
@@ -0,0 +1,818 @@
+/// Utilities for sorting constant lists of string literals.
+///
+/// Examples where these are useful:
+/// - Sorting `__all__` in the global scope,
+/// - Sorting `__slots__` or `__match_args__` in a class scope
+use std::borrow::Cow;
+use std::cmp::Ordering;
+
+use ruff_python_ast as ast;
+use ruff_python_codegen::Stylist;
+use ruff_python_parser::{lexer, Mode, Tok};
+use ruff_python_trivia::leading_indentation;
+use ruff_source_file::Locator;
+use ruff_text_size::{Ranged, TextRange, TextSize};
+
+use is_macro;
+use itertools::{izip, Itertools};
+
+/// An enumeration of the various kinds of sequences for which Python has
+/// [display literals](https://docs.python.org/3/reference/expressions.html#displays-for-lists-sets-and-dictionaries).
+///
+/// (I'm aware a set isn't actually a "sequence",
+/// *but* for our purposes it's conceptually a sequence,
+/// since in terms of the AST structure it's almost identical
+/// to tuples/lists.)
+///
+/// Whereas lists, dicts and sets are always parenthesized
+/// (e.g. lists always start with `[` and end with `]`),
+/// single-line tuples *can* be unparenthesized.
+/// We keep the original AST node around for the
+/// Tuple variant so that this can be queried later.
+#[derive(Debug)]
+pub(super) enum SequenceKind<'a> {
+    List,
+    Set,
+    Tuple(&'a ast::ExprTuple),
+}
+
+impl SequenceKind<'_> {
+    fn surrounding_parens(&self, source: &str) -> (&str, &str) {
```

Comment:
Nit, but could safe someone lifetime issues further down
```suggestion
    fn surrounding_parens(&self, source: &str) -> (&'static str, &'static str) {
```